riptide-metrics/src/main/java/org/zalando/riptide/metrics/MetricsPlugin.java

```diff
@@ -48,23 +80,54 @@ public RequestExecution prepare(final RequestArguments arguments, final RequestE
     @AllArgsConstructor
     private final class Measurement {
 
-        private final Stopwatch stopwatch = Stopwatch.createStarted();
+        private final long startTime = clock.monotonicTime();
         private final RequestArguments arguments;
 
-        void record(final ClientHttpResponse response) throws Exception {
-            stopwatch.stop();
+        void record(final @Nullable ClientHttpResponse response,
+                @SuppressWarnings("unused") @Nullable final Throwable e) {
+
+            final long endTime = clock.monotonicTime();
 
-            final Timer timer = getTimer(response);
-            final long duration = stopwatch.elapsed(MILLISECONDS);
+            final Iterable<Tag> tags = concat(defaultTags, tags(arguments, response));
+            final Timer timer = registry.timer(metricName, tags);
 
-            timer.record(duration, MILLISECONDS);
+            final long duration = endTime - startTime;
+            timer.record(duration, NANOSECONDS);
         }
 
-        private Timer getTimer(final ClientHttpResponse response) throws Exception {
-            final String metricName = generator.generate(arguments, response);
-            return registry.timer(metricName);
+    }
+
+    private Iterable<Tag> tags(final RequestArguments arguments, @Nullable final ClientHttpResponse response) {
+        return Arrays.asList(
+                Tag.of("method", method(arguments)),
+                Tag.of("uri", uri(arguments)),
+                Tag.of("status", status(response)),
+                Tag.of("clientName", client(arguments))
+        );
+    }
+
+    private String method(final RequestArguments arguments) {
+        return arguments.getMethod().name();
+    }
+
+    private String uri(final RequestArguments arguments) {
+        return firstNonNull(arguments.getUriTemplate(), arguments.getRequestUri().getPath());
+    }
+
+    private String status(@Nullable final ClientHttpResponse response) {
+        if (response == null) {
+            return "CLIENT_ERROR";
         }
 
+        try {
+            return String.valueOf(response.getRawStatusCode());
+        } catch (final IOException e) {
+            return "IO_ERROR";
```

Comment:
Any idea how to cover/enforce this?

Reply:
Or we just let the exception propagate and don't measure in these cases